numpy/_core/src/multiarray/arraytypes.c.src

```diff
@@ -1029,18 +1023,17 @@ _copy_and_return_void_setitem(_PyArray_LegacyDescr *dstdescr, char *dstdata,
 }
 
 static int
-VOID_setitem(PyObject *op, void *input, void *vap)
+VOID_setitem(PyArray_Descr *descr_, PyObject *op, char *ip)
 {
-    char *ip = input;
-    PyArrayObject *ap = vap;
-    int itemsize = PyArray_ITEMSIZE(ap);
+    _PyArray_LegacyDescr *descr = (_PyArray_LegacyDescr *)descr_;
```

Comment:
More out of curiosity, but why is this needed? (It was there already, so I guess just that some aspects cannot be retrieved by our macro API yet?)